### PR TITLE
WIP Infra documentation

### LINKED ISSUE #223: Document the setup of the "robocat" cluster
# Expected Behavior
Everything that is installed in the "robocat" cluster needs to be documented and automated, so that it may be possible to recreate the "robocat" repo from scratch at any time.

The work lays the ground for testing CD of services and configurations, as well as for testing of infra changes. Once we have the cluster setup fully automated, we'll be able to test infra changes by allocating a new cluster, setting it up, and apply the change from the PR.

# Actual Behavior
So far I created the cluster via the cloud console. 
The spec of the cluster shall be documented.

This issue can be considered closed once the documentation structure is done, and the initial setup has been documented. Any extra setup will be handled in dedicated issues.

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Restructure a bit the infra documentation.
Add detailed documentation about the "dogfooding" cluster.
Add documentation about the "robocat" cluster.
Fix old documentation here and there.

Fixes #223

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: docs/infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+The infra system relies on several different kubernetes clusters, three are
+static and the rest are dynamic (provisioned on demand).
+
+- [*prow*](../prow/README.md): Prow, Boskos and Tekton run in this cluster.
+  This cluster runs resources defined in the `prow` folder. CI Jobs that only
+  require a container run in the `test-pods` namespace of this cluster.
+- [*dogfooding*](./dogfooding.md): Tekton runs in this cluster. This cluster is
+  setup with [resources](../tekton/README.md#resources-for-cicd) from the
+  `tekton` folder, plus a few [secrets](./dogfooding.md#secrets).
+- [*robocat*](./robocat.md): This cluster is our test bed for continuous
```
Comment (vdemeester) [inline docs/infrastructure.md]: :heart_eyes_cat:

**Comment by bobcatfish** [comment]
(im just trying out the mario bot)

/mario test

**Comment by bobcatfish** [comment]
hm i guess it only responds to build commands. womp womp

**Comment by bobcatfish** [comment]
just changed the secrets, hopefully this works 🤞

**Comment by bobcatfish** [comment]
errrrrg

**Comment by bobcatfish** [comment]
yay i fixed the mario github cred at least XD

**Comment by bobcatfish** [comment]
/mario build tekton/images/test-runner test-runner:test-christie

**Comment by bobcatfish** [comment]
HAH while using this PR to experiment, I realized that this does some of what i was trying to get toward with https://github.com/tektoncd/plumbing/pull/384

@afrittoli can we merge this asap?

**Comment by vdemeester** [comment]
@afrittoli needs some rebasing and conflicts fixing :upside_down_face:

**Comment by bobcatfish** [comment]
Sorry I made this so much harder by creating all the conflicts @afrittoli :(

**Comment by bobcatfish** [comment]
I've rebased these docs and dealt with some conflicts over in https://github.com/tektoncd/plumbing/pull/444

**Pull request closed without merging**